jbpm-case-mgmt/jbpm-case-mgmt-api/src/main/java/org/jbpm/casemgmt/api/CaseService.java

```diff
@@ -338,6 +338,16 @@
     Collection<CommentInstance> getCaseComments(String caseId, CommentSortBy sortBy, QueryContext queryContext) throws CaseNotFoundException;
 
     /**
+     * Returns all case comments sorted with given sortBy
+     * @param caseId unique case id in the format PREFIX-GENERATED_ID as described on startCase method
+     * @param page defines the specific page index to query
+     * @param pageSize defines how many items per page
+     * @return a specific page of comments added to given case
+     * @throws CaseNotFoundException thrown in case case was not found with given id
+     */
+    Collection<CommentInstance> getCaseComments(String caseId, int page, int pageSize, QueryContext queryContext) throws CaseNotFoundException;
```

Comment:
QueryContext already carries pagination information so there is no need for new method